plugins/modules/vmware_dvs_portgroup.py

```diff
@@ -237,372 +263,1059 @@
-    portgroup_name: vlan-trunk-portrgoup
-    switch_name: dvSwitch
-    vlan_id: 1-1000, 1005, 1100-1200
-    vlan_trunk: True
+    portgroup: vlan-trunk-portgroup
+    switch: dvSwitch
+    vlan_trunk_range:
+      - 1-1000
+      - 1005
+      - 1100-1200
     num_ports: 120
-    portgroup_type: earlyBinding
     state: present
   delegate_to: localhost
 
-- name: Create private vlan portgroup
-  vmware_dvs_portgroup:
+- name: Create pvlan portgroup
+  community.vmware.vmware_dvs_portgroup:
     hostname: '{{ vcenter_hostname }}'
     username: '{{ vcenter_username }}'
     password: '{{ vcenter_password }}'
-    portgroup_name: private-vlan-portrgoup
-    switch_name: dvSwitch
-    vlan_id: 1001
-    vlan_private: True
+    portgroup: vlan-123-portgroup
+    switch: dvSwitch
+    private_vlan_id: 123
     num_ports: 120
-    portgroup_type: earlyBinding
     state: present
   delegate_to: localhost
 
-- name: Create no-vlan portgroup
+- name: Create no-vlan portgroup with no uplinks
   community.vmware.vmware_dvs_portgroup:
     hostname: '{{ vcenter_hostname }}'
     username: '{{ vcenter_username }}'
     password: '{{ vcenter_password }}'
-    portgroup_name: no-vlan-portrgoup
-    switch_name: dvSwitch
-    vlan_id: 0
+    portgroup: no-vlan-portgroup
+    switch: dvSwitch
     num_ports: 120
-    portgroup_type: earlyBinding
+    teaming:
+      active_uplinks: []
+      standby_uplinks: []
     state: present
   delegate_to: localhost
 
-- name: Create vlan portgroup with all security and port policies
+- name: Create vlan portgroup with all settings
   community.vmware.vmware_dvs_portgroup:
     hostname: '{{ vcenter_hostname }}'
     username: '{{ vcenter_username }}'
     password: '{{ vcenter_password }}'
-    portgroup_name: vlan-123-portrgoup
-    switch_name: dvSwitch
+    portgroup: vlan-123-portgroup
+    switch: dvSwitch
     vlan_id: 123
     num_ports: 120
-    portgroup_type: earlyBinding
-    state: present
-    network_policy:
-      promiscuous: true
-      forged_transmits: true
-      mac_changes: true
-    port_policy:
+    port_binding: static
+    port_allocation: elastic
+    security:
+      promiscuous_mode: no
+      mac_changes: no
+      forged_transmits: no
+    teaming:
+      load_balancing: loadbalance_srcid
+      network_failure_detection: link_status_only
+      notify_switches: True
+      failback: True
+      active_uplinks: [ 'Uplink 1' ]
+      standby_uplinks: [ 'Uplink 2' ]
+    advanced:
+      port_config_reset_at_disconnect: true
       block_override: true
-      ipfix_override: true
-      live_port_move: true
+      netflow_override: true
       network_rp_override: true
-      port_config_reset_at_disconnect: true
       security_override: true
       shaping_override: true
       traffic_filter_override: true
       uplink_teaming_override: true
       vendor_config_override: true
       vlan_override: true
+    state: present
   delegate_to: localhost
 '''
 
+RETURN = """
+result:
```

Comment:
`result` is internally used by Ansible Core Engine. Since we are here, can you please change this to something meaningful?

Reply:
ok. done